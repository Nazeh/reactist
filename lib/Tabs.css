.reactist.tabs__header {
  display: flex;
  border-bottom: 1px solid #ececec;
}
.reactist.tabs__header--item {
  display: block;
  margin: 0 20px 0 0;
  padding: 10px 0;
  font-size: 0.875rem;
  color: #808080;
  font-weight: 400;
  line-height: 1.7;
  text-decoration: none;
  box-sizing: border-box;
}
.reactist.tabs__header--item.active {
  font-size: 0.875rem;
  color: #3F82EF;
  font-weight: 400;
  line-height: 1.7;
}
.reactist.tabs__header--item.active:after {
  display: block;
  content: '';
  background-color: #3F82EF;
  height: 3px;
  border-radius: 3px 3px 0 0 ;
  position: relative;
  top: 10px;
}
.reactist.tabs__header--item.disabled:hover {
  cursor: not-allowed;
}
.reactist.tabs--spreadlayout .tabs__header {
  display: flex;
  border-bottom: 0;
}
.reactist.tabs--spreadlayout .tabs__header--item {
  text-align: center;
  flex: 1;
  border-bottom: 1px solid #ececec;
  margin: 0;
}
.reactist.tabs--spreadlayout .tabs__header--item.active {
  font-size: 0.875rem;
  color: #3F82EF;
  font-weight: 400;
  line-height: 1.7;
  border-bottom: 3px solid #3F82EF;
}
.reactist.tabs--spreadlayout .tabs__header--item:hover {
  text-decoration: none;
}


/*# sourceMappingURL=Tabs.map*/