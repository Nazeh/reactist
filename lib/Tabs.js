!function(e,t){"object"==typeof exports&&"object"==typeof module?module.exports=t(require("react")):"function"==typeof define&&define.amd?define("reactist",["react"],t):"object"==typeof exports?exports.reactist=t(require("react")):e.reactist=t(e.React)}("undefined"!=typeof self?self:this,function(e){return function(e){function t(n){if(r[n])return r[n].exports;var o=r[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}var r={};return t.m=e,t.c=r,t.d=function(e,r,n){t.o(e,r)||Object.defineProperty(e,r,{configurable:!1,enumerable:!0,get:n})},t.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return t.d(r,"a",r),r},t.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},t.p="",t(t.s=192)}({1:function(t,r){t.exports=e},10:function(e,t,r){var n,o;/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/
!function(){"use strict";function r(){for(var e=[],t=0;t<arguments.length;t++){var n=arguments[t];if(n){var o=typeof n;if("string"===o||"number"===o)e.push(n);else if(Array.isArray(n)&&n.length){var i=r.apply(null,n);i&&e.push(i)}else if("object"===o)for(var s in n)a.call(n,s)&&n[s]&&e.push(s)}}return e.join(" ")}var a={}.hasOwnProperty;void 0!==e&&e.exports?(r.default=r,e.exports=r):(n=[],void 0!==(o=function(){return r}.apply(t,n))&&(e.exports=o))}()},192:function(e,t,r){"use strict";function n(e){return e&&e.__esModule?e:{default:e}}function o(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}function a(e,t){if(!e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return!t||"object"!=typeof t&&"function"!=typeof t?e:t}function i(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function, not "+typeof t);e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,enumerable:!1,writable:!0,configurable:!0}}),t&&(Object.setPrototypeOf?Object.setPrototypeOf(e,t):e.__proto__=t)}Object.defineProperty(t,"__esModule",{value:!0}),t.Tab=t.Tabs=void 0;var s=function(){function e(e,t){for(var r=0;r<t.length;r++){var n=t[r];n.enumerable=n.enumerable||!1,n.configurable=!0,"value"in n&&(n.writable=!0),Object.defineProperty(e,n.key,n)}}return function(t,r,n){return r&&e(t.prototype,r),n&&e(t,n),t}}();r(193);var u=r(1),c=n(u),f=r(2),l=n(f),p=r(10),d=n(p),b=function(e){function t(e,r){o(this,t);var n=a(this,(t.__proto__||Object.getPrototypeOf(t)).call(this,e,r));n._switchActiveTab=function(e,t){n.setState(function(){return{activeTabIndex:t}}),n.props.onChange&&n.props.onChange(e.props.value)},n._renderTabLinks=function(e){return e.map(function(e,t){var r=e.props,o=r.title,a=r.disabled,i=e.props.value||t,s=(0,d.default)("reactist tabs__header--item",{disabled:a,active:t===n.state.activeTabIndex});return c.default.createElement("a",{className:s,href:"",key:i,onClick:function(r){r.preventDefault(),a||n._switchActiveTab(e,t)}},o)})};var i=e.defaultTab,s=e.onChange,u=c.default.Children.toArray(n.props.children),f=i||0===i;if((f||s)&&u.find(function(e){return!e.props.value&&0!==e.props.value}))throw new Error('(Tab) Missing property: all Tab must have "value" set if "defaultTab" or "onChange" is used');if(f){var l=u.findIndex(function(e){return e.props.value===i});if(-1===l)throw new Error('(Tabs) Unable to find Tab with the matching defaultTab value "'+i+'"');n.state={activeTabIndex:l}}else n.state={activeTabIndex:0};return n}return i(t,e),s(t,[{key:"render",value:function(){var e=c.default.Children.toArray(this.props.children),t=e[this.state.activeTabIndex]||e[0]||null,r=(0,d.default)("reactist","tabs",this.props.spreadLayout?"tabs--spreadlayout":null);return c.default.createElement("div",{className:r},c.default.createElement("div",{className:"reactist tabs__header"},this._renderTabLinks(e)),c.default.createElement("div",{className:"reactist tabs__body"},t))}}]),t}(c.default.Component);b.displayName="Tabs",b.propTypes={defaultTab:l.default.oneOfType([l.default.string,l.default.number]),spreadLayout:l.default.bool,onChange:l.default.func,children:l.default.any},b.defaultProps={spreadLayout:!1};var h=function(e){var t=e.children,r=e.className;return c.default.createElement("div",{className:(0,d.default)("reactist tabs__tab",r)},t)};h.displayName="Tab",h.defaultProps={disabled:!1},h.propTypes={value:l.default.oneOfType([l.default.string,l.default.number]),title:l.default.string.isRequired,disabled:l.default.bool,className:l.default.string,children:l.default.any},t.Tabs=b,t.Tab=h},193:function(e,t,r){var n=r(194);"string"==typeof n&&(n=[[e.i,n,""]]);var o={hmr:!0};o.transform=void 0,r(4)(n,o),n.locals&&(e.exports=n.locals)},194:function(e,t,r){t=e.exports=r(3)(!1),t.push([e.i,".reactist.tabs__header{display:flex;border-bottom:1px solid #ececec}.reactist.tabs__header--item{display:block;margin:0 20px 0 0;padding:10px 0;font-size:.875rem;color:#b3b3b3;font-weight:400;line-height:1.7;text-decoration:none;box-sizing:border-box}.reactist.tabs__header--item.active{font-size:.875rem;color:#3f82ef;font-weight:400;line-height:1.7;border-bottom:3px solid #3f82ef}.reactist.tabs__header--item.disabled:hover{cursor:not-allowed}.reactist.tabs--spreadlayout .tabs__header{display:flex;border-bottom:0}.reactist.tabs--spreadlayout .tabs__header--item{text-align:center;flex:1;border-bottom:1px solid #ececec;margin:0}.reactist.tabs--spreadlayout .tabs__header--item.active{font-size:.875rem;color:#3f82ef;font-weight:400;line-height:1.7;border-bottom:3px solid #3f82ef}.reactist.tabs--spreadlayout .tabs__header--item:hover{text-decoration:none}",""])},2:function(e,t,r){e.exports=r(6)()},3:function(e,t){function r(e,t){var r=e[1]||"",o=e[3];if(!o)return r;if(t&&"function"==typeof btoa){var a=n(o);return[r].concat(o.sources.map(function(e){return"/*# sourceURL="+o.sourceRoot+e+" */"})).concat([a]).join("\n")}return[r].join("\n")}function n(e){return"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(e))))+" */"}e.exports=function(e){var t=[];return t.toString=function(){return this.map(function(t){var n=r(t,e);return t[2]?"@media "+t[2]+"{"+n+"}":n}).join("")},t.i=function(e,r){"string"==typeof e&&(e=[[null,e,""]]);for(var n={},o=0;o<this.length;o++){var a=this[o][0];"number"==typeof a&&(n[a]=!0)}for(o=0;o<e.length;o++){var i=e[o];"number"==typeof i[0]&&n[i[0]]||(r&&!i[2]?i[2]=r:r&&(i[2]="("+i[2]+") and ("+r+")"),t.push(i))}},t}},4:function(e,t,r){function n(e,t){for(var r=0;r<e.length;r++){var n=e[r],o=b[n.id];if(o){o.refs++;for(var a=0;a<o.parts.length;a++)o.parts[a](n.parts[a]);for(;a<n.parts.length;a++)o.parts.push(f(n.parts[a],t))}else{for(var i=[],a=0;a<n.parts.length;a++)i.push(f(n.parts[a],t));b[n.id]={id:n.id,refs:1,parts:i}}}}function o(e,t){for(var r=[],n={},o=0;o<e.length;o++){var a=e[o],i=t.base?a[0]+t.base:a[0],s=a[1],u=a[2],c=a[3],f={css:s,media:u,sourceMap:c};n[i]?n[i].parts.push(f):r.push(n[i]={id:i,parts:[f]})}return r}function a(e,t){var r=v(e.insertInto);if(!r)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var n=g[g.length-1];if("top"===e.insertAt)n?n.nextSibling?r.insertBefore(t,n.nextSibling):r.appendChild(t):r.insertBefore(t,r.firstChild),g.push(t);else if("bottom"===e.insertAt)r.appendChild(t);else{if("object"!=typeof e.insertAt||!e.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var o=v(e.insertInto+" "+e.insertAt.before);r.insertBefore(t,o)}}function i(e){if(null===e.parentNode)return!1;e.parentNode.removeChild(e);var t=g.indexOf(e);t>=0&&g.splice(t,1)}function s(e){var t=document.createElement("style");return e.attrs.type="text/css",c(t,e.attrs),a(e,t),t}function u(e){var t=document.createElement("link");return e.attrs.type="text/css",e.attrs.rel="stylesheet",c(t,e.attrs),a(e,t),t}function c(e,t){Object.keys(t).forEach(function(r){e.setAttribute(r,t[r])})}function f(e,t){var r,n,o,a;if(t.transform&&e.css){if(!(a=t.transform(e.css)))return function(){};e.css=a}if(t.singleton){var c=m++;r=y||(y=s(t)),n=l.bind(null,r,c,!1),o=l.bind(null,r,c,!0)}else e.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(r=u(t),n=d.bind(null,r,t),o=function(){i(r),r.href&&URL.revokeObjectURL(r.href)}):(r=s(t),n=p.bind(null,r),o=function(){i(r)});return n(e),function(t){if(t){if(t.css===e.css&&t.media===e.media&&t.sourceMap===e.sourceMap)return;n(e=t)}else o()}}function l(e,t,r,n){var o=r?"":n.css;if(e.styleSheet)e.styleSheet.cssText=_(t,o);else{var a=document.createTextNode(o),i=e.childNodes;i[t]&&e.removeChild(i[t]),i.length?e.insertBefore(a,i[t]):e.appendChild(a)}}function p(e,t){var r=t.css,n=t.media;if(n&&e.setAttribute("media",n),e.styleSheet)e.styleSheet.cssText=r;else{for(;e.firstChild;)e.removeChild(e.firstChild);e.appendChild(document.createTextNode(r))}}function d(e,t,r){var n=r.css,o=r.sourceMap,a=void 0===t.convertToAbsoluteUrls&&o;(t.convertToAbsoluteUrls||a)&&(n=x(n)),o&&(n+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(o))))+" */");var i=new Blob([n],{type:"text/css"}),s=e.href;e.href=URL.createObjectURL(i),s&&URL.revokeObjectURL(s)}var b={},h=function(e){var t;return function(){return void 0===t&&(t=e.apply(this,arguments)),t}}(function(){return window&&document&&document.all&&!window.atob}),v=function(e){var t={};return function(r){if(void 0===t[r]){var n=e.call(this,r);if(n instanceof window.HTMLIFrameElement)try{n=n.contentDocument.head}catch(e){n=null}t[r]=n}return t[r]}}(function(e){return document.querySelector(e)}),y=null,m=0,g=[],x=r(5);e.exports=function(e,t){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");t=t||{},t.attrs="object"==typeof t.attrs?t.attrs:{},t.singleton||"boolean"==typeof t.singleton||(t.singleton=h()),t.insertInto||(t.insertInto="head"),t.insertAt||(t.insertAt="bottom");var r=o(e,t);return n(r,t),function(e){for(var a=[],i=0;i<r.length;i++){var s=r[i],u=b[s.id];u.refs--,a.push(u)}e&&n(o(e,t),t);for(var i=0;i<a.length;i++){var u=a[i];if(0===u.refs){for(var c=0;c<u.parts.length;c++)u.parts[c]();delete b[u.id]}}}};var _=function(){var e=[];return function(t,r){return e[t]=r,e.filter(Boolean).join("\n")}}()},5:function(e,t){e.exports=function(e){var t="undefined"!=typeof window&&window.location;if(!t)throw new Error("fixUrls requires window.location");if(!e||"string"!=typeof e)return e;var r=t.protocol+"//"+t.host,n=r+t.pathname.replace(/\/[^\/]*$/,"/");return e.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(e,t){var o=t.trim().replace(/^"(.*)"$/,function(e,t){return t}).replace(/^'(.*)'$/,function(e,t){return t});if(/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/)/i.test(o))return e;var a;return a=0===o.indexOf("//")?o:0===o.indexOf("/")?r+o:n+o.replace(/^\.\//,""),"url("+JSON.stringify(a)+")"})}},6:function(e,t,r){"use strict";var n=r(7),o=r(8),a=r(9);e.exports=function(){function e(e,t,r,n,i,s){s!==a&&o(!1,"Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types")}function t(){return e}e.isRequired=e;var r={array:e,bool:e,func:e,number:e,object:e,string:e,symbol:e,any:e,arrayOf:t,element:e,instanceOf:t,node:e,objectOf:t,oneOf:t,oneOfType:t,shape:t,exact:t};return r.checkPropTypes=n,r.PropTypes=r,r}},7:function(e,t,r){"use strict";function n(e){return function(){return e}}var o=function(){};o.thatReturns=n,o.thatReturnsFalse=n(!1),o.thatReturnsTrue=n(!0),o.thatReturnsNull=n(null),o.thatReturnsThis=function(){return this},o.thatReturnsArgument=function(e){return e},e.exports=o},8:function(e,t,r){"use strict";function n(e,t,r,n,a,i,s,u){if(o(t),!e){var c;if(void 0===t)c=new Error("Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings.");else{var f=[r,n,a,i,s,u],l=0;c=new Error(t.replace(/%s/g,function(){return f[l++]})),c.name="Invariant Violation"}throw c.framesToPop=1,c}}var o=function(e){};e.exports=n},9:function(e,t,r){"use strict";e.exports="SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED"}})});
//# sourceMappingURL=Tabs.map