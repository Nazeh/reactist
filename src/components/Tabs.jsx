import './styles/tabs.less'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class Tabs extends React.Component {
    constructor(props, context) {
        super(props, context)

        const { defaultTab, onChange } = props
        const children = React.Children.toArray(this.props.children)

        const hasDefault = defaultTab || defaultTab === 0
        if (hasDefault || onChange) {
            const missing = children.find(c => !c.props.value)
            if (missing) throw new Error('(Tab) Missing property: all Tab must have "value" set if "defaultTab" or "onChange" is used')
        }

        if (hasDefault) {
            const i = children.findIndex((x) => x.props.value === defaultTab)
            if (i === -1)  throw new Error(`(Tabs) Unable to find Tab with the matching defaultTab value "${defaultTab}"`)

            this.state = { activeTabIndex: i }
        } else {
            this.state = { activeTabIndex: 0 }
        }
    }

    _switchActiveTab = (tab, i) => {
        this.setState(() => ({ activeTabIndex: i }))
        if (this.props.onChange) this.props.onChange(tab.props.value)
    }

    _renderTabLinks = (tabs) => {
        return tabs.map((t, i) => {
            const { title, disabled } = t.props
            const value = t.props.value || i
            const className = classNames('reactist tabs__header--item',
                { disabled, active: i === this.state.activeTabIndex }
            )

            return (
                <a
                    className={ className }
                    href=''
                    key={ value }
                    onClick={ (event) => {
                        event.preventDefault()
                        if (!disabled) {
                            this._switchActiveTab(t, i)
                        }
                    } }
                >
                    { title }
                </a>
            )
        })
    }

    render() {
        // ensures that single or no child components don't throw
        const children = React.Children.toArray(this.props.children)
        const activeTab = children[this.state.activeTabIndex] || children[0] || null

        const cls = classNames(
            'reactist',
            'tabs',
            this.props.spreadLayout ? 'tabs--spreadlayout' : null
        )

        return (
            <div className={ cls }>
                <div className='reactist tabs__header'>
                    { this._renderTabLinks(children) }
                </div>
                <div className='reactist tabs__body'>
                    { activeTab }
                </div>
            </div>
        )
    }
}
Tabs.displayName = 'Tabs'

Tabs.propTypes = {
    /** selects the tab whose value prop matches this prop */
    defaultTab: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    /** Whether the tabs should take all available space and distribute it evenly or use the minimum required **/
    spreadLayout: PropTypes.bool,
    /** Callback for tab change event. Tab value will be passed */
    onChange: PropTypes.func
}
Tabs.defaultProps = {
    spreadLayout: false
}

const Tab = ({ children, className }) => (
    <div className={ classNames('reactist tabs__tab', className) }>
        { children }
    </div>
)
Tab.displayName = 'Tab'
Tab.defaultProps = {
    disabled: false
}
Tab.propTypes = {
    /* It assigns a value to the tab so that it can be selected by the Tabs.*/
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),

    /** Title of the tab. */
    title: PropTypes.string.isRequired,
    /** Disabled tabs can't be selected. */
    disabled: PropTypes.bool,
    /** Additional css class applied to Tab. */
    className: PropTypes.string
}


export { Tabs, Tab }
