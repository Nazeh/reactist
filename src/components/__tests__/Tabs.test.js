import React from 'react'
import { shallow } from 'enzyme'

import { Tabs, Tab } from '../Tabs'

describe('Tabs', () => {
    it('renders without children', () => {
        const tabs = shallow(<Tabs />)
        expect(tabs).toMatchSnapshot()
    })

    it('renders with single children', () => {
        const tabs = shallow(
            <Tabs>
                <Tab title="Tab 1">Content</Tab>
            </Tabs>
        )
        expect(tabs).toMatchSnapshot()
    })

    it('renders Tab without children', () => {
        const tab = shallow(<Tab title="Tab" />)
        expect(tab).toMatchSnapshot()
    })

    it('activates the defaultTab on initial render', () => {
        const tabs = shallow(
            <Tabs defaultTab={3}>
                <Tab value={1} title="Tab 1">
                    Content
                </Tab>
                <Tab value={2} title="Tab 2">
                    Content
                </Tab>
                <Tab value={3} title="Tab 3">
                    Content
                </Tab>
            </Tabs>
        )

        const tab3 = tabs.find('.reactist_tabs__header--item').at(2)
        expect(tab3.hasClass('active')).toBe(true)
    })

    it('activates a tab upon clicking on it', () => {
        const tabs = shallow(
            <Tabs>
                <Tab title="Tab 1">
                    <div id="tab1" />
                </Tab>
                <Tab title="Tab 2">
                    <div id="tab2" />
                </Tab>
            </Tabs>
        )

        expect(tabs.find('#tab1')).toHaveLength(1)
        expect(tabs.find('#tab2')).toHaveLength(0)

        const tab2 = tabs.find('.reactist_tabs__header--item').at(1)
        tab2.simulate('click', { preventDefault: jest.fn() })

        expect(tabs.find('#tab1')).toHaveLength(0)
        expect(tabs.find('#tab2')).toHaveLength(1)
    })

    it('invoke onChange() with tab value when a tab is activated', () => {
        const f = jest.fn()

        const tabs = shallow(
            <Tabs onChange={f}>
                <Tab title="Tab 1" value="1" />
                <Tab title="Tab 2" value="2" />
            </Tabs>
        )

        expect(f).not.toHaveBeenCalled()

        const tab2 = tabs.find('.reactist_tabs__header--item').at(1)
        tab2.simulate('click', { preventDefault: jest.fn() })

        expect(f).toHaveBeenCalledWith('2')
        expect(f).toHaveBeenCalledTimes(1)
    })

    it('does not activate a disabled tab upon clicking on it', () => {
        const tabs = shallow(
            <Tabs>
                <Tab title="Tab 1">
                    <div id="tab1" />
                </Tab>
                <Tab title="Tab 2" disabled>
                    <div id="tab2" />
                </Tab>
            </Tabs>
        )

        expect(tabs.find('#tab1')).toHaveLength(1)
        expect(tabs.find('#tab2')).toHaveLength(0)

        const tab2 = tabs.find('.reactist_tabs__header--item').at(1)
        tab2.simulate('click', { preventDefault: jest.fn() })

        expect(tabs.find('#tab1')).toHaveLength(1)
        expect(tabs.find('#tab2')).toHaveLength(0)
    })

    it('raise error if no tab with the match defaultTab value is found', () => {
        expect(() => {
            shallow(
                <Tabs defaultTab={3}>
                    <Tab value={1} title="Tab 1">
                        Content
                    </Tab>
                    <Tab value={2} title="Tab 2">
                        Content
                    </Tab>
                    <Tab value={4} title="Tab 4">
                        Content
                    </Tab>
                </Tabs>
            )
        }).toThrow()
    })

    it('accepts 0 (zero) as valid value for Tab and defaultTab', () => {
        expect(() => {
            shallow(
                <Tabs defaultTab={0}>
                    <Tab value={0} title="Tab 0">
                        Content
                    </Tab>
                    <Tab value={1} title="Tab 1">
                        Content
                    </Tab>
                </Tabs>
            )
        }).not.toThrow()
    })
})
