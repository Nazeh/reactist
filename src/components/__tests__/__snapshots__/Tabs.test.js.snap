// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders Tab without children 1`] = `
<div
  className="reactist_tabs__tab"
/>
`;

exports[`Tabs renders with single children 1`] = `
<div
  className="reactist_tabs"
>
  <div
    className="reactist_tabs__header"
  >
    <a
      className="reactist_tabs__header--item active"
      href=""
      key="0"
      onClick={[Function]}
    >
      Tab 1
    </a>
  </div>
  <div
    className="reactist_tabs__body"
  >
    <Tab
      disabled={false}
      key=".0"
      title="Tab 1"
    >
      Content
    </Tab>
  </div>
</div>
`;

exports[`Tabs renders without children 1`] = `
<div
  className="reactist_tabs"
>
  <div
    className="reactist_tabs__header"
  />
  <div
    className="reactist_tabs__body"
  />
</div>
`;
